core/src/main/java/io/cloudevents/core/impl/BaseCloudEventBuilder.java

```diff
@@ -140,6 +141,15 @@ public SELF withExtension(@Nonnull String key, @Nonnull OffsetDateTime value) {
         return self;
     }
 
+    @Override
+    public CloudEventBuilder withExtension(@Nonnull String key, @Nonnull byte[] value) {
```

Comment:
In the builder tests, can you add one case where you build a cloudevent with a binary extension and then you retrieve such extension and check the type is still `byte[]`?